### ISSUE TITLE
Helsingin Sanomat comics at https://www.hs.fi/sarjakuvat/

### ISSUE THREAD

**Issue by ExTechOp** [sailfishos-applications/daily-comics#103]
**Daily Comics VERSION** (Daily Comics → [Top pulley] About): ***`<1.9.95>`***

*\<Please do name only a single comic / cartoon in each "fix comic" suggestion!  The only exception is when a format change by a comics provider affects multiple comics retrieved from this provider.\>*

#### NECESSARY DATA
- Name of the comic<br />
  ***\<Fingerpori\>***
- Denote the URL of a source page which allows for extracting a URL of a single image of the most recent, published strip by a RegEx (see next bullet point).  This source page may be identical to the home-page of the comic.<br />
  ***https://www.hs.fi/fingerpori/***
- An idea for a RegEx (regular expression), containing a sub-expression which exactly matches the image path (either as a full URL or a path relative to the source page URL from the prior bullet point) to be extracted from the content of the source page.<br />
  ***`<sorry, none>`***
- If such a single RegEx is not feasible, please design a JavaScript code snippet, which extracts the URL of the image of the most recent comic strip starting from the content retrieved via the URL provided in the second bullet point of this section "NECESSARY DATA".
  ```
  <JavaScript code>
  ```

#### ADDITIONAL INFORMATION
*\<Please consider thoroughly which other pieces of information might be relevant.\>*

Three HS comics stopped working at the same time: simultaneously: Fingerpori, Viivi ja Wagner, and FOK_IT

**Comment by Olf0** [comment]
Technically this should bot be hard: Take a look by `Inspect` (left mouse button) a comic strip or look at the page source.

An exemplary comic list item looks like this (using https://www.hs.fi/fingerpori/ on 2024-10-12):
```
<span class="date">Lauantaina, 12.10.</span>
</span>
<a href="/fingerpori/car-2000010758312.html">
<figure class="cartoon image scroller" itemscope itemtype="http://schema.org/ImageObject" id="cartoon-2000010758312">
<img id="image-8145639"
class="lazyload lazyloadable-image "
data-alt=""
data-simple-src="//images.sanoma-sndp.fi/d46c8befaf65903b8f0ad5cc969d5bd2.jpg/normal/320.jpg"
data-srcset="//images.sanoma-sndp.fi/d46c8befaf65903b8f0ad5cc969d5bd2.jpg/normal/1920.jpg 1920w"
/>
```

Hence the RegEx should be something like (untested): `/data-srcset="(\/\/[^" ]+)/`
Mind that the extracted link from the first hit (only) has to be prepended with an `https:` before returning it.

The same should be applicable for all three, "Fingerpori", "Viivi ja Wagner" and "FOK_IT".

P.S.: Please also check the generic metadata (authors etc.) is still correct.

P.P.S.: I currently absolutely lack the time (at least for the next couple of months) for implementing and testing this for these three HS-comics, but it is should be easy by the help of this guide: https://github.com/sailfishos-applications/daily-comics/blob/devel/info/comic_addition.md
<sub>(Well, requires a little JavaScript and RexEx know-how, which can be acquired by the many tutorials and references in the WWW on these two topics.</sub>

But I will sure try to give good advice if your stuck and ask something specific, review a resulting PR, and ultimately create a new release, when your PR became merged.

Hence I am looking forward to your contribution!

**Comment by ExTechOp** [comment]
I've now created a minimal PR #104 for Fingerpori only as a proof of concept. If you find it acceptable, I'll make similar changes to the other mentioned HS strips.

**Comment by Olf0** [comment]
Extant HS-comics in Daily Comics are fixed by PR #105.

It would be nice to add the (three, AFAICS) HS-comics [in the table above](https://github.com/sailfishos-applications/daily-comics/issues/103#issuecomment-2408326702) which are not yet supported by Daily Comics.

**Comment by Olf0** [comment]
Plug-ins for the remaining three comic strips from HS are implemented by PR #108.
